src/main/java/com/google/sps/utils/QuizTimingPropertiesUtils.java

```diff
@@ -107,22 +119,42 @@ public Boolean isQuizOutdated(Long current_quiz_time) {
     public String getNewQuestion(Entity game_entity, DatastoreService datastore) {
         Random rand = new Random();
         int rand_number = rand.nextInt(quiz_questions.size());
-        Key game_entity_key;
- 
+
         try {
-            game_entity_key = game_entity.getKey();
+            game_entity.setProperty("quiz_timestamp", System.currentTimeMillis());
         } catch(NullPointerException e) {
             log.log(Level.SEVERE, "Null value given for parameter {0}", game_entity);
             return null;
         } 
- 
-        datastore.delete(game_entity_key);
-            
-        Entity update_game_entity = new Entity(game_entity_key);
-        update_game_entity.setProperty("quiz_timestamp", System.currentTimeMillis());
-        update_game_entity.setProperty("quizQuestion", quiz_questions.get(rand_number));
-        datastore.put(update_game_entity);
-        return (update_game_entity.getProperty("quizQuestion")).toString();
+
+        game_entity.setProperty("quizQuestion", quiz_questions.get(rand_number));
+        datastore.put(game_entity);
+        return (game_entity.getProperty("quizQuestion")).toString();
     }
 
+    //Gives the user 20 points if they have taken a quiz
+    public boolean giveUserQuizTakenPoints(boolean userQuizStatus, Entity currentUser, DatastoreService datastore) {
+        if(currentUser == null) {
+            log.log(Level.SEVERE, "Given a null {0}", currentUser);
+            return false;
+        }
+
+        if(datastore == null) {
+            log.log(Level.SEVERE, "Given a null {0}", datastore);
+            return false;
+        }
+
+        if(userQuizStatus) {
+            try {
+                currentUser.setProperty("score", ((Number) currentUser.getProperty("score")).intValue() + 20);
+            } catch (NullPointerException e) {
+                log.log(Level.SEVERE, "User Entity has no 'score' property");
+                return false;
```

Comment:
If user has no score property, maybe we can be clever and assign them one here? I know the user *should* already have it, but if they don't it would make life easier if we create the property with a default value of whatever the points score is.